calm_adapter/calm_adapter/src/main/scala/uk/ac/wellcome/calm_adapter/CalmXmlResponse.scala

```diff
@@ -96,13 +96,21 @@ case class CalmSummaryResponse(val root: Elem, retrievedAt: Instant)
       .flatMap(_.childWithTag("SummaryList"))
       .flatMap(_.childWithTag("Summary"))
       .map(_ \ "_")
-      .flatMap { node =>
-        val data = node.map(child => child.label -> child.text).toMap
-        data
-          .get("RecordID")
-          .map(id => Right(CalmRecord(id, data, retrievedAt)))
-          .getOrElse(Left(new Exception("RecordID not found")))
+      .flatMap { nodes =>
+        val data = toMapping(nodes)
+        data.get("RecordID") match {
+          case Some(List(id)) => Right(CalmRecord(id, data, retrievedAt))
+          case Some(_)        => Left(new Exception("Multiple RecordIDs found"))
+          case None           => Left(new Exception("RecordID not found"))
```

Comment:
Ah OK - I was going to do this at the transformer level?

Reply:
You are correct for all fields apart from `RecordID`: we need a unique identifier for the VHS key, so need to parse this here.